go/engine/kex2_provisionee.go

```diff
@@ -93,8 +94,12 @@ func (e *Kex2Provisionee) Run(ctx *Context) error {
 		panic("empty secret")
 	}
 
+	var nctx context.Context
+	nctx, e.kex2Cancel = context.WithCancel(ctx.GetNetContext())
+	defer e.kex2Cancel()
```

Comment:
The doc says to call cancel at the end regardless:

https://godoc.org/golang.org/x/net/context

to release resources.

It should be a no-op as it is just closing the Done channel.